src/plugins/intel_gpu/tests/test_cases/gather_gpu_test.cpp

```diff
@@ -8,10 +8,334 @@
+
+        set_values(input0, dat);
+        set_values(input1, ind);
+
+        topology reorder_topo;
+        reorder_topo.add(input_layout("input0", input0->get_layout()));
+        reorder_topo.add(input_layout("input1", input1->get_layout()));
+        reorder_topo.add(reorder("reorder0", "input0", fmt[0], T_dat_dt));
+        reorder_topo.add(reorder("reorder1", "input1", fmt[1], T_ind_dt));
+        reorder_topo.add(gather("gather",
+                                "reorder0",
+                                "reorder1",
+                                axis,
+                                ov::Shape(shape_out.begin(), shape_out_end_iter),
+                                batch_dim,
+                                true));
+        reorder_topo.add(reorder("reorder2", "gather", format::type::bfwzyx, T_dat_dt));
+
+        network reorder_network(engine, reorder_topo);
+        reorder_network.set_input_data("input0", input0);
+        reorder_network.set_input_data("input1", input1);
+
+        auto reorder_output = reorder_network.execute().at("reorder2").get_memory();
+        cldnn::mem_lock<T_dat> reorder_output_ptr(reorder_output, get_test_stream());
+
+        topology planar_topo;
+        planar_topo.add(input_layout("input0", input0->get_layout()));
+        planar_topo.add(input_layout("input1", input1->get_layout()));
+        planar_topo.add(gather("gather",
+                               "input0",
+                               "input1",
+                               axis,
+                               ov::Shape(shape_out.begin(), shape_out_end_iter),
+                               batch_dim,
+                               true));
+
+        network planar_network(engine, planar_topo);
+        planar_network.set_input_data("input0", input0);
+        planar_network.set_input_data("input1", input1);
+
+        auto planar_output = planar_network.execute().at("gather").get_memory();
+        cldnn::mem_lock<T_dat> planar_output_ptr(planar_output, get_test_stream());
+
+        EXPECT_TRUE(!memcmp(reorder_output_ptr.data(), planar_output_ptr.data(), mult(shape_out) * sizeof(T_dat)));
+    }
+};
+
+using gather8_test_4d_f16i32 = gather8_test_4d<FLOAT16, int, data_types::f16, data_types::i32>;
+TEST_P(gather8_test_4d_f16i32, gather8_test_4d_f16i32) {}
+INSTANTIATE_TEST_SUITE_P(gather8_4d_bd0_d4_i1,
+                         gather8_test_4d_f16i32,
+                         testing::Combine(testing::Values(0),
+                                          testing::Values(0),  //[batch_dim,dim(dict))
+                                          testing::Values(format::type::bfyx),
+                                          testing::Values(format::type::bfyx),
+                                          testing::Values(std::array<int, 4>{5, 44, 7, 8}),
+                                          testing::Values(std::array<int, 4>{4, 1, 1, 1})));
```

Comment:
I think you can merge many `INSTANTIATE` into smaller list with `testing::Values`

Reply:
Passing multiple values to testing::Values() generates testcases as Cartesian product, which conflicts with feedback that there are too many test cases.

Reply:
Added some disabled testcases for future usage.